src/sql/parts/notebook/notebookActions.ts

```diff
@@ -412,6 +412,7 @@ export class AttachToDropdown extends SelectBox {
 				let attachToConnections = this.values;
 				if (!connection) {
 					this.loadAttachToDropdown(this.model, this.getKernelDisplayName());
+					this.doChangeContext();
```

Comment:
This may work, but I feel like this is a patch over something else. Logic is already super complicated as it is :(.

When we click cancel on the add new connection dialog, why can't we just... do nothing?

Reply:
If we just ignore without changing the context on cancel, it will set to the previous connection. Assume that you selected sql connection with Spar kernel, it will set back to sql connection which shouldn't happen. Now that we have integrated unified connection, we should show 'Select Connection' on cancelling dialog. This would a corner case scenario.

Reply:
Also why do we want to have an invalid connection message here? if you just cancel something I feel like the state should just remain (except that select connection should be selected).

Anyone else want to chime in here :)?

Reply:
Changed code so that it won't show error message upon clicking cancel from connection dialog. Let me know if any one has different thoughts.